### DISCUSSION TITLE
RuntimeError: CUDA out of memory

### DISCUSSION THREAD

**Discussion by backlow** [hlky/stable-diffusion-webui#1419] [Q&A]
It seems looking fine except getting this error message on final when i execute webui.cmd 

RuntimeError: CUDA out of memory. Tried to allocate 58.00 MiB (GPU 0; 2.00 GiB total capacity; 1.61 GiB already allocated; 0 bytes free; 1.67 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

How can i fix it?

**Comment by waxnssg** [comment]
I have a same problem.

**Comment by sherry0429** [comment]
buy a new graphics card will solve this problem, my old card only 4GB gpu memory, so this question occur.  a graphics card with more than 12 GB gpu memory will solve this. it's not too expensive now.

by the way, if you still want to run, don't want to paid money, you could choose run it on cpu, but it's very slow.
here have a gist you could try:

```
import torch
from torch import autocast
from diffusers import StableDiffusionPipeline
from diffusers import StableDiffusionImg2ImgPipeline
import os
import requests
from PIL import Image
from io import BytesIO
os.environ['CUDA_VISIBLE_DEVICES'] = ""
os.environ['NVIDIA_VISIBLE_DEVICES'] = "all"

model_id = "CompVis/stable-diffusion-v1-4"
device = "cpu"


def text2img(prompt):
    pipe = StableDiffusionPipeline.from_pretrained(
        model_id,
        use_auth_token=True,
        resume_download=True
        # torch_dtype=torch.float16, revision="fp16"
    )
    pipe = pipe.to(device)
    image = pipe(prompt).images[0]
    image.save("astronaut_rides_horse.png")


def img2img(url):
    pipe = StableDiffusionImg2ImgPipeline.from_pretrained(
        model_id,
        use_auth_token=True,
        resume_download=True
        # torch_dtype=torch.float16, revision="fp16"
    )
    if url.startswith('http'):
        response = requests.get(url)
        init_image = Image.open(BytesIO(response.content)).convert("RGB")
        init_image = init_image.resize((512, 512))
    else:
        init_image = Image.open(BytesIO(open(url, 'rb').read())).convert("RGB")
        init_image = init_image.resize((512, 512))
    prompt = "A fantasy landscape, trending on artstation"
    images = pipe(prompt=prompt, init_image=init_image, strength=0.75, guidance_scale=7.5).images
    images[0].save("fantasy_landscape.png")


if __name__ == '__main__':
    text2img("A vertical painting of anime girl, long and black hair, blue eyes, in armor, equip sword and shield")
    img2img("https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg")
```

if this answer help you, please accept it, thank you @backlow

**Comment by easonoob** [comment]
You can check the boxes for optimised mode and float16 in the settings tab to reduce cuda memory usage 😄

**Reply by easonoob** [reply]
Also set batch size to 1 and choose a lower resolution like 512x512, you can use upscalers such as RealESRGAN and LDSR to upscale the img later
